### ISSUE TITLE
React Native federatedSignin with Google is not working

### ISSUE THREAD

**Issue by alpeshgaglani** [aws-amplify/amplify-js#3177]
I cannot get Google signin to work on react native with Auth.federatedSignin({provider:"Google"});

We have what I think, is a very common scenario: Our requirement is to allow google/facebook signin for web and react native. We are using Cognito Userpools. 
Starting with web, I added "https://mydomain.io/login/" for signin in our userpool client, got client creds from google, and fed them to the Google Identity Provider in the userpool. This setup works just fine - I'm able to signin and create new users, and even invoke api in api gateway just fine.

For react native, we setup url scheme "myapp" in Info.plist for iOS to enable deep linking. I also added "myapp://authorize/" to the signin redirects in userpool app client. So our Callback URL in UserPool looks like "myapp://authorize/, "https://mydomain.io/login/". I then call Auth.federatedSignin({provider:"Google"}) and setup Hub listeners for the auth channel. On making this call, the client opens the browser, shows the hosted UI, signs the user in and returns back to our app. However, things just break down from there. 

This is what I see in the logs:
```
DEBUG] 21:46.677 OAuth - Starting code flow with myapp://authorize/?code=guid&state=longstring
[DEBUG] 21:46.678 Hub - Dispatching to auth with {"event":"codeFlow","data":{},"message":"Retrieving tokens from https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token"}
[DEBUG] 21:46.679 Analytics - on hub capsule auth{"event":"codeFlow","data":{},"message":"Retrieving tokens from https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token"}
[DEBUG] 21:46.680 Hub - Dispatching to auth with {"event":"codeFlow","data":{},"message":"Retrieving tokens from https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token"}
{"channel":"auth","payload":{"event":"codeFlow","data":{},"message":"Retrieving tokens from https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token"},"source":"Auth","patternInfo":[]}
{"event":"codeFlow","data":{},"message":"Retrieving tokens from https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token"}
[DEBUG] 21:46.683 OAuth - Calling token endpoint: https://mydomain.auth.us-west-2.amazoncognito.com/oauth2/token with{"grant_type":"authorization_code","code":"guid","client_id":"7pa0s7cc3tpolm26g3eno9g04r","redirect_uri":"myapp://authorize/","code_verifier":"long string"}
[DEBUG] 21:46.869 AuthClass - Error in cognito hosted auth response{}
[DEBUG] 21:46.870 Hub - Dispatching to auth with {"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"}
[DEBUG] 21:46.874 Analytics - on hub capsule auth{"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"}
[DEBUG] 21:46.875 Credentials - getting credentials
[DEBUG] 21:46.876 Credentials - picking up credentials
[DEBUG] 21:46.878 Credentials - getting new cred promise
[DEBUG] 21:46.879 Credentials - checking if credentials exists and not expired
[DEBUG] 21:46.880 Credentials - need to get a new credential or refresh the existing one
[DEBUG] 21:46.880 AuthClass - Getting current user credentials
[DEBUG] 21:46.881 Hub - Dispatching to auth with {"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"}
[DEBUG] 21:46.882 Hub - Dispatching to auth with {"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"}
{"channel":"auth","payload":{"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"},"source":"Auth","patternInfo":[]}
{"event":"signIn_failure","data":{},"message":"The OAuth response flow failed"}
[DEBUG] 21:46.883 Hub - Dispatching to auth with {"event":"cognitoHostedUI_failure","data":{},"message":"A failure occurred when returning to the Cognito Hosted UI"}
[DEBUG] 21:46.883 Analytics - on hub capsule auth{"event":"cognitoHostedUI_failure","data":{},"message":"A failure occurred when returning to the Cognito Hosted UI"}
[DEBUG] 21:46.884 Hub - Dispatching to auth with {"event":"cognitoHostedUI_failure","data":{},"message":"A failure occurred when returning to the Cognito Hosted UI"}
{"channel":"auth","payload":{"event":"cognitoHostedUI_failure","data":{},"message":"A failure occurred when returning to the Cognito Hosted UI"},"source":"Auth","patternInfo":[]}
{"event":"cognitoHostedUI_failure","data":{},"message":"A failure occurred when returning to the Cognito Hosted UI"}
```

This is what my amplify.configure looks:
```
{"Auth": {
		"userPoolId": "userpoolid",
		"identityPoolId": "identitypoolid",
		"region": "us-west-2",
		"userPoolWebClientId": "webclientid",
		"oauth": {
			"domain": "mydomain.auth.us-west-2.amazoncognito.com",
			"redirectSignIn": "myapp://authorize/",
			"redirectSignOut": "myapp://logout/",
			"scope": [
				"profile",
				"openid",
				"email"
			],
			"responseType": "code"
		}
	}
}
```

Is there any documentation I can look at to help me unblock myself?

**Comment by manueliglesias** [comment]
Hi @alpeshgaglani 

Looks like there's an error happening when exchanging the code for tokens, this is done via an http POST request. From your logs, the error looks empty (`[DEBUG] 21:46.869 AuthClass - Error in cognito hosted auth response{}`). Is there a way you could inspect the network calls? (e.g. by using [reactotron](https://github.com/infinitered/reactotron))

**Comment by alpeshgaglani** [comment]
> Hi @alpeshgaglani
> 
> Looks like there's an error happening when exchanging the code for tokens, this is done via an http POST request. From your logs, the error looks empty (`[DEBUG] 21:46.869 AuthClass - Error in cognito hosted auth response{}`). Is there a way you could inspect the network calls? (e.g. by using [reactotron](https://github.com/infinitered/reactotron))

On more debugging, it looks like it fails when I have debugger attached. I can sign in just fine when debugger is detached. Any clues as to why?

**Comment by haverchuck** [comment]
@alpeshgaglani If you are using Chrome debugging, the JavaScript[ runs within Chrome itself](https://facebook.github.io/react-native/docs/javascript-environment#javascript-runtime).  This could potentially explain some unexpected behavior like this. 

I am closing this issue, but if the answer is not satisfactory please feel free to open a new issue with additional detail.
